pkg/helpers/helpers.go

```diff
@@ -0,0 +1,25 @@
+package helpers
+
+import (
+	"fmt"
+	"time"
+)
+
+func PollImmediate(interval time.Duration, timeout time.Duration, condition func() (bool, error)) error {
+
+	var totalTime time.Duration = 0
+
+	for totalTime < timeout {
+		cond, err := condition()
+		if cond {
+			return nil
+		}
+		if err != nil {
+			return err
+		}
+		time.Sleep(interval)
+		totalTime += interval
+	}
+
+	return fmt.Errorf("timed out")
```

Comment:
A typed error would enable the caller to reliably determine whether the failure here was due to a timeout or an error returned from the condition func. See https://github.com/kubernetes/apimachinery/blob/276a8a7530a30f42c82b1f0fa2261a905130bf97/pkg/util/wait/wait.go#L672 for example.